include/arc_utilities/dijkstras.hpp

```diff
@@ -735,24 +791,19 @@ namespace arc_dijkstras
                 const AstarNode top_node = queue.top();
                 queue.pop();
                 // Check if the node has already been discovered
-                const auto explored_itr = explored.find(top_node.Index());
-                // We have already been here and it was cheaper then
-                if ((explored_itr != explored.end()) && (top_node.CostToCome() >= explored_itr->second.second))
-                {
-                    continue;
-                }
+                const auto explored_itr = explored.find(top_node.GraphIndex());
                 // We have not been here before, or it is cheaper now
-                else
+                if (!((explored_itr != explored.end()) && (top_node.CostToCome() >= explored_itr->second.second)))
```

Comment:
I think we might be okay for this sort of boolean thing. C++ seems to be fast fail/succeed for &&, ||.
http://en.cppreference.com/w/cpp/language/eval_order
See item 6 in rules.

I know I was taught that `if (a != null && f(a->data))` was the correct way to use pointers.